pydata_sphinx_theme/__init__.py

```diff
@@ -28,6 +28,13 @@ def update_config(app, env):
                 "Use `search-field.html` in `navbar_end` template list instead."
             )
         )
+    if not isinstance(theme_options.get("icon_links", []), list):
+        raise ExtensionError(
+            (
+                "`icon_links` must be a list of dictionaries, you provided "
+                f"type {type(theme_options.get("icon_links"))}."
```

Comment:
updated this so that we don't explicitly tell them the keys to use, in case it changes we don't wanna have to maintain it :-)